addons/website_event/views/snippets/s_events.xml

```diff
@@ -0,0 +1,107 @@
+            </figcaption>
+        </a>
+    </figure>
+</template>
+
+<template id="dynamic_filter_template_event_event_card" name="Card Layout" priority="10">
+    <div t-foreach="records" t-as="data" class="pb32 w-100 s_events_event">
+        <t t-set="record" t-value="data['_record']"/>
+        <div class="card shadow-sm">
+            <a class="s_events_event_cover" t-att-href="data['call_to_action_url']">
+                <t t-call="website.record_cover">
+                    <t t-set="_record" t-value="record"/>
+                    <t t-set="_resize_height" t-value="256"/>
+                    <t t-set="_resize_width" t-value="256"/>
+
+                    <div class="s_events_event_date position-absolute bg-white shadow-sm text-dark">
+                        <span t-field="record.with_context(tz=record.date_tz).date_begin" t-options="{'format': 'LLL'}" class="s_events_event_month"/>
+                        <span t-field="record.with_context(tz=record.date_tz).date_begin" t-options="{'format': 'dd'}" class="s_events_event_day"/>
+                    </div>
+                </t>
+            </a>
+            <div class="card-body p-3">
+                <div t-if="is_sample" class="h5 o_ribbon_right bg-primary text-uppercase">Sample</div>
+                <h5 class="mb-0 text-truncate" t-field="record.name"/>
+                <time itemprop="startDate" t-att-datetime="record.date_begin">
+                    <span t-field="record.with_context(tz=record.date_tz).date_begin" t-options="{'date_only': 'true', 'format': 'long'}"/> -
+                    <span t-field="record.with_context(tz=record.date_tz).date_begin" t-options="{'time_only': 'true', 'format': 'short'}"/>
+                </time>
+                <div itemprop="location" t-field="record.address_id" t-options="{'widget': 'contact', 'fields': ['city'], 'no_marker': 'true'}"/>
+            </div>
+        </div>
+    </div>
+</template>
+
+<!--SNIPPET OPTIONS -->
+
+<template id="s_events_options" inherit_id="website.snippet_options">
+    <xpath expr="." position="inside">
+        <t t-call="website_event.s_dynamic_snippet_options_template">
+            <t t-set="snippet_name" t-value="'event_upcoming_snippet'"/>
+            <t t-set="snippet_selector" t-value="'.s_event_upcoming_snippet'"/>
+            <t t-set="grouping_message">Items per row</t>
+        </t>
+    </xpath>
+</template>
+
+<template id="s_dynamic_snippet_options_template" inherit_id="website.s_dynamic_snippet_options_template">
+    <xpath expr="//we-select[@data-name='filter_opt']" position="after">
+        <we-many2many t-if="snippet_name == 'event_upcoming_snippet'"
+                      string="Event Tags"
+                      data-name="event_tag_opt"
+                      data-model="event.event"
+                      data-m2o-field="tag_ids"
+                      data-domain='[["category_id.website_published", "=", true], ["color", "!=", false], ["color", "!=", 0]]'
```

Comment:
Same remark for domain, if we can concatenate both color leaves into a single leaf, would make it easier to read